File: src/components/xl_storage/xl_types_golang/xl_meta_v1_object.go
package xl_types_golang

//go:generate msgp

// A xlMetaV1Object represents `xl.meta` metadata header.
type XlMetaV1Object struct {
	Version string   `json:"version"` // Version of the current `xl.meta`.
	Format  string   `json:"format"`  // Format of the current `xl.meta`.
	Stat    StatInfo `json:"stat"`    // Stat of the current object `xl.meta`.
	// Erasure coded info for the current object `xl.meta`.
	Erasure ErasureInfo `json:"erasure"`
	// MinIO release tag for current object `xl.meta`.
	Minio struct {
		Release string `json:"release"`
	} `json:"minio"`
	// Metadata map for current object `xl.meta`.
	Meta map[string]string `json:"meta,omitempty"`
	// Captures all the individual object `xl.meta`.
	Parts []ObjectPartInfo `json:"parts,omitempty"`

	// Dummy values used for legacy use cases.
	VersionID string `json:"versionId,omitempty"`
	DataDir   string `json:"dataDir,omitempty"` // always points to "legacy"
}

File: src/components/xl_storage/xl_types/xl_meta_v1_object.rs
use serde::{Deserialize, Serialize};
use std::collections::HashMap;

use super::{erasure_info::ErasureInfo, object_part_info::ObjectPartInfo, stat_info::StatInfo};

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct XlMetaV1Object {
    pub version: String,
    pub format: String,
    pub stat: StatInfo,
    pub erasure: ErasureInfo,
    pub minio: MinioInfo,
    #[serde(skip_serializing_if = "Option::is_none")]
    pub meta: Option<HashMap<String, String>>,
    #[serde(skip_serializing_if = "Option::is_none")]
    pub parts: Option<Vec<ObjectPartInfo>>,
    #[serde(rename = "versionId", skip_serializing_if = "Option::is_none")]
    pub version_id: Option<String>,
    #[serde(rename = "dataDir", skip_serializing_if = "Option::is_none")]
    pub data_dir: Option<String>,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct MinioInfo {
    pub release: String,
}
